### DISCUSSION TITLE
Error setting up my local development environment on Ubuntu

### DISCUSSION THREAD

**Discussion by xaviibaez** [anyproto/contributors#47] [Any-Block & Anytype-Heart]
Hello all! Im planning to start contributing on code anytype-ts repo. 
Which version on nodejs is prefered? I tried a lot of but none are working on doing 'npm install -D'. 
I'm currently using node 18.16.0 and 16.20.1. I tried those deleting node_modules and installing via command again and this error on electron builder happens both ways. I'm following the README.md and may I need some other previous software?
Im using ubuntu 22.04.
![image](https://github.com/anyproto/contributors/assets/25080554/6032a9ac-1b62-4fef-8c3d-42f925a15a67)

If you need more info about my system or software im using please tell me here :)

**Comment by fuksman** [comment]
@anyproto/desktop

**Reply by xaviibaez** [reply]
Hey! thanks for the reply. Is this the anytype desktop localization repo I need? https://github.com/anyproto/l10n-anytype-ts I'm not completely understaing your reply. Thanks!

**Reply by fuksman** [reply]
Hi! I've just mentioned the desktop team of Anytype, let's wait until they answer :)

**Reply by xaviibaez** [reply]
oh thanks!

**Comment by ra3orblade** [comment]
You should install libsecret, if you read the message carefully it says libbsecret package is not found, it's a dependency for keytar. https://github.com/anyproto/anytype-ts check readme

**Reply by xaviibaez** [reply]
my bad, Ill try again. Thanks

**Reply by xaviibaez** [reply]
I tried again and it worked! 
Maybe we can update anytype-ts and anytype-heart with these suggestions, shall I create a PR?
- Consider adding into './update.sh' step 'apt-get install jq'
- This one may be not necessary but  consider adding a consideration about defining env variable 'usr/local/go/bin' into '/etc/enviroment' to define permanently access go command everywhere.
- On making a build on anytype-heart with 'make install-dev-js' after 'apt install protobuf-compiler libprotoc-dev' step consider doing the  steps on https://stackoverflow.com/questions/60388933/installation-of-protoc-gen-grpc-web && https://github.com/grpc/grpc-web#code-generator-plugin. It solves a problem I encountered.
- On anytype-ts doing 'npm run dist:(mac|win|linux)' I faced two missing dependencies: 'sudo snap install snapcraft --classic' and 'sudo apt-get install rpm' solved my problem.

**Reply by xaviibaez** [reply]
Lastly on doing 'npm run dist:(mac|win|linux)'  I'm facing these error:

![image](https://github.com/anyproto/contributors/assets/25080554/3dd9060b-cd51-4c2d-add5-7c24e6c816c6)

I looked for the 'ERR_ELECTRON_BUILDER_CANNOT_EXECUTE' and found that wasn't solved by the creator. I may be wrong. Did you have this problem and how do you solve it? Thanks.

**Reply by ra3orblade** [reply]
I did not solve it, I do not build linux locally. And you do not need to build it either to launch the app, you can run it as a local dev server via `SERVER_PORT=8080 npm run start:dev`.
Concerning dependencies, actually update.sh was not designed for community to build middleware - it's a script for CI to download release for building purposes, we always build middleware from source internally.

**Reply by xaviibaez** [reply]
Thnaks, I'll try later with the info you give me.

**Reply by xaviibaez** [reply]
@ra3orblade it worked fine! Thanks!
